release/calculate.go

```diff
@@ -59,47 +59,43 @@ func Calculate(finder ModuleFinder, tags git.ModuleTags, config repotools.Config
 		var latestVersion string
 		var hasChanges bool
 
-		latestVersion, ok := tags.Latest(moduleDir)
+		latestVersion, ok := tags.Latest(module.Path())
 		if ok {
-			startTag, err := git.ToModuleTag(moduleDir, latestVersion)
+			startTag, err := git.ToModuleTag(module.Path(), latestVersion)
 			if err != nil {
 				log.Fatalf("failed to convert module path and version to tag: %v", err)
 			}
 
-			changes, err := git.Changes(finder.Root(), startTag, "HEAD", moduleDir)
+			changes, err := git.Changes(finder.Root(), startTag, "HEAD", module.Path())
 			if err != nil {
 				log.Fatalf("failed to get git changes: %v", err)
 			}
 
-			subModulePaths := repositoryModules[moduleDir]
-
-			ignoredModulePaths := make([]string, 0, len(subModulePaths)+len(repositoryModuleTombstonePaths))
-			ignoredModulePaths = append(ignoredModulePaths, subModulePaths...)
-
-			if len(repositoryModuleTombstonePaths) > 0 {
-				ignoredModulePaths = append(ignoredModulePaths, repositoryModuleTombstonePaths...)
-				// IsModuleChanged expects the provided list of ignored modules paths to be sorted
-				sort.Strings(ignoredModulePaths)
-			}
-
-			hasChanges, err = gomod.IsModuleChanged(moduleDir, ignoredModulePaths, changes)
+			hasChanges, err = gomod.IsModuleChanged(module, changes)
 			if err != nil {
 				return nil, fmt.Errorf("failed to determine module changes: %w", err)
 			}
 
 			if !hasChanges {
-				// Check if any of the submodules have been "carved out" of this module since the last tagged release
-				for _, subModuleDir := range subModulePaths {
-					if _, ok := tags.Latest(subModuleDir); ok {
+				// TODO filter out tombstone modules?
```

Comment:
This poses a good question.....in theory we could put a model under a path that was formally occupied by some parent tombstoned module....but that might get really weird. I don't know if we need to solve it now, but likely we may just want to implement a check that prevents creation of nested modules under tumbstone paths all together.

Reply:
This is one thing that was left out of the change. There is no annotation for tombstone, modules. `No-release` modules are not included at all since they didn't need to be processed.  I've started testing of this change with aws-sdk-go-v2 to verify its behavior. I'll investigate if the no-release modules need to be enumerated, and if tombstones need additional annotations